src/openfermion/utils/_sparse_tools_test.py

```diff
@@ -108,6 +108,60 @@ def test_qubit_operator_sparse_n_qubits_not_specified(self):
             qubit_operator_sparse(QubitOperator('X1')).A,
             expected.A))
 
+    def test_get_linear_qubit_operator_x(self):
+        vec = numpy.array([1, 2, 3, 4])
+        matvec_expected = numpy.array([2, 1, 4, 3])
+
+        self.assertTrue(numpy.allclose(
+            get_linear_qubit_operator(QubitOperator('X1')) * vec,
+            matvec_expected))
+
+    def test_get_linear_qubit_operator_y(self):
+        vec = numpy.array([1, 2, 3, 4], dtype=complex)
+        matvec_expected = 1.0j * numpy.array([-2, 1, -4, 3], dtype=complex)
+
+        self.assertTrue(numpy.allclose(
+            get_linear_qubit_operator(QubitOperator('Y1')) * vec,
+            matvec_expected))
+
+    def test_get_linear_qubit_operator_z(self):
+        vec = numpy.array([1, 2, 3, 4])
+        matvec_expected = numpy.array([1, 2, -3, -4])
+
+        self.assertTrue(numpy.allclose(
+            get_linear_qubit_operator(QubitOperator('Z0'), 2) * vec,
+            matvec_expected))
+
+    def test_get_linear_qubit_operator_z3(self):
+        vec = numpy.array(
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16])
+        matvec_expected = numpy.array(
+            [1, -2, 3, -4, 5, -6, 7, -8, 9, -10, 11, -12, 13, -14, 15, -16])
+
+        self.assertTrue(numpy.allclose(
+            get_linear_qubit_operator(QubitOperator('Z3')) * vec,
+            matvec_expected))
+
+    def test_get_linear_qubit_operator_zx(self):
+        """Testing with multiple factors."""
+        vec = numpy.array([1, 2, 3, 4])
+        matvec_expected = numpy.array([2, 1, -4, -3])
+
+        self.assertTrue(numpy.allclose(
+            get_linear_qubit_operator(QubitOperator('Z0 X1')) * vec,
+            matvec_expected))
+
+    def test_get_linear_qubit_operator_multiple_terms(self):
+        """Testing with multiple terms."""
+        qubit_operator = QubitOperator(((0, 'Z'), (1, 'X')), 10.0) + 2.0 * \
+            QubitOperator('Y2')
+
+        vec = numpy.array([1, 2, 3, 4, 5, 6, 7, 8])
+        matvec_expected = 10 * numpy.array([3, 4, 1, 2, -7, -8, -5, -6]) + \
+            2.j * numpy.array([-2, 1, -4, 3, -6, 5, -8, 7], dtype=complex)
+
+        self.assertTrue(numpy.allclose(
+            get_linear_qubit_operator(qubit_operator) * vec, matvec_expected))
```

Comment:
Instead of hard coding all the tests, let's have at least one that checks consistency between our other methods for building qubit operators and the LinearOperator defined either by their action on a random normalized vector, or their action on the identity matrix (for small cases) to check the matrix representations.  That way we also detect any convention changes between different features that happen by accident later.

Reply:
Good point, added compare test with another function.